torchmultimodal/utils/generate.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import warnings
+from typing import Any, Optional
+
+import torch
+from torch import nn, Tensor
+from torch.nn import functional as F
+
+from torchmultimodal.utils.attention import get_causal_attention_mask
+
+
+class Generation:
```

Comment:
High-level q: why make this a class and not a function/script?

Reply:
I was on the fence about it. We can turn this back into a function but in general "Generation" can involve several techniques other than multinomial sampling (which is what we are doing now) such that `Class` will be more appropriate for sharing of common attributes / methods.